src/rhsmlib/services/register.py

```diff
@@ -43,22 +43,24 @@ def register(self, org, activation_keys=None, environment=None, force=None, name
         # signature we want to consider that an error.
         if kwargs:
             raise exceptions.ValidationError(_("Unknown arguments: %s") % kwargs.keys())
-
+        refreshToken = None
         syspurpose = syspurposelib.read_syspurpose()
         role = role or syspurpose.get('role', '')
         addons = addons or syspurpose.get('addons', [])
         usage = usage or syspurpose.get('usage', '')
         service_level = service_level or syspurpose.get('service_level_agreement', '')
+        if token:
+            refreshToken = token[0]
```

Comment:
When length of token is 0, then it will raise exception. Please add here check for zero length of token. Something like:
```python
if token and len(token) > 0:
```